src/test/test_dir.c

```diff
@@ -3830,6 +3864,83 @@ mock_get_options(void)
   return mock_options;
 }
 
+/**
+ * Test dirauth_get_b64_digest_bw_file.
+ * This function should be near the other bwauth functions, but it needs
+ * mock_get_options, that is only defined here.
+ */
+
+static void
+test_dir_bwauth_bw_file_digest256(void *arg)
+{
+  (void)arg;
+  const char *content =
+    "1541171221\n"
+    "node_id=$68A483E05A2ABDCA6DA5A3EF8DB5177638A27F80 "
+    "master_key_ed25519=YaqV4vbvPYKucElk297eVdNArDz9HtIwUoIeo0+cVIpQ "
+    "bw=760 nick=Test time=2018-05-08T16:13:26\n";
+
+  char *fname = tor_strdup(get_fname("V3BandwidthsFile"));
+  /* Initialize to a wrong digest. */
+  uint8_t digest[DIGEST256_LEN] = "01234567890123456789abcdefghijkl";
+
+  /* Digest of an empty string */
+  char digest_empty_str[DIGEST256_LEN] = "01234567890123456789abcdefghijkl";
+  crypto_digest256(digest_empty_str, "", 0, DIGEST_SHA256);
+
+  /* Digest of the content */
+  char digest_expected2[DIGEST256_LEN] = "01234567890123456789abcdefghijkl";
+  crypto_digest256(digest_expected2, content, sizeof(content), DIGEST_SHA256);
+  /* Logs 1DC64B095F76C89980B23C99D06B749D6B46DAA9C270E4FA1E027605CBC02A39 */
```

Comment:
This hash calculation is wrong.
sizeof(content) is sizeof(const char *) which is 8 on 64-bit platforms.

I can match this hash with:
```
$ printf "15411712" | shasum -a 256
1dc64b095f76c89980b23c99d06b749d6b46daa9c270e4fa1e027605cbc02a39  -
```